msprime/demography.py

```diff
@@ -954,16 +957,26 @@ def insert_extra_populations(self, tables):
                     "tree sequence",
                     IncompletePopulationMetadataWarning,
                 )
+            left_out = False
             for population in self.populations[len(tables.populations) :]:
                 md = {}
                 if name_in_metadata:
                     md["name"] = population.name
                 if description_in_metadata:
                     md["description"] = population.description
-                # TODO we could also try to merge in the ``extra_metadata``
-                # depending on whether `additional_properties` is set,
-                # but it doesn't really seem worth it.
-                tables.populations.add_row(md)
+                for k in population.extra_metadata:
+                    if k in properties or additional_properties:
+                        md[k] = population.extra_metadata[k]
+                    else:
+                        left_out = True
```

Comment:
It would probably be helpful to tell the user what the left out fields are, how about ``left_out.append(k)`` and then insert ``left_out`` into the warning message?